### ISSUE TITLE
Some requests (stream based) never ends and block the queue

### ISSUE THREAD

**Issue by Verhov** [bda-research/node-crawler#371]
### Summary:
I parsed millions domains and faced issue what some domain request (stream based) can permanently block the queue.
I found one: [https://goldfm.nu/](https://goldfm.nu/)
It's really nice radio :smile: but totally block my pattern))

### Current behavior
I'am using timeout but looks it not work pretty correctly, callback never fired in this case:
```
_crawler = new Crawler({
      timeout: 9000,
      retries: 1,
      retryTimeout: 1000,
      debug: true,
      callback: (error, res, done) => {
          ...
          done()
      }
})

_crawler.queue([{ uri: 'https://goldfm.nu' }])
```
![image](https://user-images.githubusercontent.com/1572713/102702375-03f51e00-4273-11eb-81ea-25fc498cfec3.png)

### Issue
Definitely it because of this request starts media stream and node-crawler tried to get it all... request always in pending state.
![image](https://user-images.githubusercontent.com/1572713/102702441-ba590300-4273-11eb-901f-4674a820a3b6.png)

### Side issues
Also as stream is arriving it increase RAM and seem will thrown 'out of memory' exception.
![image](https://user-images.githubusercontent.com/1572713/102702678-597efa00-4276-11eb-9ce0-16872fd17d94.png)

### Attempts to fix
I also tried to set accept header to html only, but it's doesn't have affect.
```headers: { Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9' },```

Currently I just skip this url as the special case, but I think it may not be unique.

### Expected behavior
Timeout should fire an error when we did not receive a response within the allotted time.

### Related issues
This issue is definitely related with [request](https://github.com/request/request) package.

- https://github.com/request/request/issues/3341

### Question
Do you have any ideas how to resolve this case?) :question:

**Comment by slienceisgolden** [comment]
have the same issue. the spider must have not only a timeout, but also a limit on the download volume

**Comment by mike442144** [comment]
Refer to my comment here: https://github.com/request/request/issues/3341
Feel free to discuss if any more questions. and hope it helps.

**Comment by Verhov** [comment]
Thanks @mike442144, but in this (crawling) context we can't blacklisted it before we face it.
And it's not very good to wait few days before the server decides to disconnect - we will receive continuous payload on both sides during this time.

I still don't know how we can identify this type of connection in advance and complete it - Iam tried to send a OPTIONS request at first, but it did't helps to detect next GET request type.

The most elegant solution in my opinion would be a some timeout, and '_response size limit_' option that @slienceisgolden mentioned would be great (incl. other pitfalls: huge docs, files, other streams etc..).

Currently not working on it, but it's still relevant.

**Comment by mike442144** [comment]
@Verhov Good idea to limit response body size, should work well in your case. Body size, also should be in the options for flexibility. Look forward to your merge request :)
